thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -630,27 +503,30 @@ class CIndexReadSlaveActivity : public CIndexReadSlaveBase
         }
 
         first = true;
-        eoi = false;
         keyedLimit = helperKeyedLimit;
         keyedLimitCount = RCMAX;
         keyedProcessed = 0;
         if (steppedExtra)
             steppingMeta.setExtra(steppedExtra);
 
-        if (partDescs.ordinality())
+        if (currentManager)
         {
-            Owned<CIndexReadHelper> indexReadHelper = new CIndexReadHelper(*this);
             if ((keyedLimit != RCMAX && (keyedLimitSkips || (helper->getFlags() & TIRcountkeyedlimit) != 0)))
-                keyedLimitCount = indexReadHelper->getCount(keyedLimit);
-            eoi = false;
-            out.setown(indexReadHelper.getClear());
+            {
+                IKeyManager *_currentManager = currentManager;
+                keyedLimitCount = getCount(keyedLimit);
+                setManager(_currentManager);
+            }
         }
         else if ((keyedLimit != RCMAX && (keyedLimitSkips || (helper->getFlags() & TIRcountkeyedlimit) != 0)))
-            keyedLimitCount = 0;            
+        {
+            eoi = false;
+            keyedLimitCount = 0;
+        }
         else
             eoi = true; // otherwise delayed until calc. in nextRow()
```

Comment:
trivial: I think this comment i probably out of date.  I'm not sure what it means.